src/Tools/dotnet-monitor/CollectionRules/ActionOptionsDependencyAnalyzer.cs

```diff
@@ -0,0 +1,269 @@
+
+                    if (!GetActionResultReference(actionAndResult, actionIndex, out CollectionRuleActionOptions dependencyOptions, out string actionResultName))
+                    {
+                        continue;
+                    }
+
+                    if (propertyDependency == null)
+                    {
+                        propertyDependency = GetOrCreateDependency(actionIndex, property, originalValue);
+                    }
+
+                    var dependency = new ActionDependency(dependencyOptions, actionResultName);
+                    propertyDependency.ActionDependencies.Add(dependency);
+                }
+            }
+        }
+
+        private PropertyDependency GetOrCreateDependency(int actionIndex, PropertyInfo propertyInfo, string originalValue)
+        {
+            if (!_dependencies.TryGetValue(actionIndex, out Dictionary<string, PropertyDependency> properties))
+            {
+                properties = new Dictionary<string, PropertyDependency>();
+                _dependencies.Add(actionIndex, properties);
+            }
+            if (!properties.TryGetValue(propertyInfo.Name, out PropertyDependency dependency))
+            {
+                dependency = new PropertyDependency(propertyInfo, originalValue);
+                properties.Add(propertyInfo.Name, dependency);
+            }
+            return dependency;
+        }
+
+        private bool GetActionResultReference(string actionReference, int actionIndex,
+            out CollectionRuleActionOptions action, out string actionResultName)
+        {
+            action = null;
+            actionResultName = null;
+
+            string[] parts = actionReference.Split(Separator);
+            if (parts.Length != 2)
+            {
+                _ruleContext.Logger.InvalidActionReference(actionReference);
+                return false;
+            }
+
+            string name = parts[0];
+            //We only check previous actions for our dependencies.
+            CollectionRuleActionOptions dependencyOptions = _ruleContext.Options.Actions.Take(actionIndex)
+                .FirstOrDefault(a => a.Name.Equals(name, StringComparison.OrdinalIgnoreCase));
+
+            if (dependencyOptions == null)
+            {
+                _ruleContext.Logger.InvalidActionReference(actionReference);
+                return false;
+            }
+
+            string resultName = parts[1];
+            if (string.IsNullOrEmpty(resultName))
+            {
+                _ruleContext.Logger.InvalidActionReference(actionReference);
+                return false;
+            }
+
+            action = dependencyOptions;
+            actionResultName = resultName;
+            return true;
+        }
+
+        private IEnumerable<PropertyInfo> GetPropertiesFromSettings(CollectionRuleActionOptions options) =>
+            options.Settings.GetType()
```

Comment:
Technically, actions are not required to have a settings instance (for example, user does not specify settings, this using all of the default values for each of the properties). However, none of the actions so far allow for a null settings property. At least add a consideration explaining why this is okay for now and the potential way to fix it.